app/resources/drafts.py

```diff
@@ -82,6 +82,13 @@ def put(self, draft_id):
         is_draft = self.check_valid_draft(draft_id)
         if is_draft is False:
             raise (BadRequest(description="Draft put requires valid draft"))
+
+        not_modified = self.etag_check(request.headers, is_draft[1])
+        if not_modified == False:
+            res = Response(response="Draft has been modified since last check", status=409,
+                           mimetype="text/html")
```

Comment:
check ? Surely it should be 'Draft has been modified since it was last read' not check